tls/s2n_config.c

```diff
@@ -130,7 +130,7 @@ static int s2n_config_init(struct s2n_config *config)
     }
 
     notnull_check(config->domain_name_to_cert_map = s2n_map_new_with_initial_capacity(1));
-    GUARD(s2n_map_complete(config->domain_name_to_cert_map));
+    GUARD_RESULT_AS_POSIX(s2n_map_complete(config->domain_name_to_cert_map));
```

Comment:
does POSIX refers to the POSIX return status? took me awhile to gather what that meant...

Reply:
Yeah: `-1/0`. I tried to think of something better but that was what I arrived at. I'm open to suggestions.

Reply:
Is this just temporary until everything returns the correct type, or is this the behavior moving forward?

I would like to be able to call `GUARD_S2N_RESULT(s2n_*)`, `GUARD_OSSL_RESULT()`, and `GUARD_POSIX_RESULT(free(ptr))`. I should know what to use based on the function name (or whether the function is in the standard library). This could aide with static analysis tooling later on.

Reply:
We will need at least the following `GUARD`s long term:

* `s2n_result` -> `s2n_result` (currently called `GUARD_RESULT` but I would imagine it would just be `GUARD` long term) This will be the most commonly used macro, after everything is converted.
* `POSIX` -> `s2n_result` (currently called `GUARD_POSIX_AS_RESULT`) This will be used when calling OS APIs (e.g. sockets, etc.). It also enables this migration to be incremental and will be used with s2n functions that still return `int`.
* `s2n_result` -> `POSIX` (currently called `GUARD_RESULT_AS_POSIX`) This will be used for preserving existing public API compatibility, which returns an `int`; either `-1` on error, `0` on ok.
* `openssl` -> `s2n_result` (currently missing - i'll need to add it) Obviously used when calling openssl APIs

I'm pretty indifferent on naming here as long as it's clear what it's doing.